compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1140,7 +1159,7 @@ public void visit(BLangSimpleVariable varNode, AnalyzerData data) {
                 dlog.error(varNode.pos, DiagnosticErrorCode.INVALID_ISOLATED_QUALIFIER_ON_MODULE_NO_INIT_VAR_DECL);
             }
 
-            if (Types.getReferredType(lhsType).tag == TypeTags.ARRAY
+            if (ownerSymTag != SymTag.TUPLE_TYPE && Types.getReferredType(lhsType).tag == TypeTags.ARRAY
```

Comment:
Why do we need this addition?

Reply:
Since tuple members are treated as variables now without this check it will throw an error when member is a closed array

Reply:
Can you give an example? If the member is an array with inferred size, we need to log an error?

Reply:
The error here is not correct right? I think we still should throw `CANNOT_INFER_SIZE_ARRAY_SIZE_FROM_THE_CONTEXT` as before

Reply:
OK. Please create an issue to improve/change the error for record fields too.

Reply:
Do we have tests for this?

Reply:
Test cases for the error exists in `testCodeAnalysisNegativeSealedArrays` in `SealedArrayTest`